conjur/user/user_controller.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+
+"""
+UserController module
+
+This module is the controller that facilitates all user actions
+"""
+
+# Builtins
+import getpass
+import logging
+import sys
+import requests
+
+# Internals
+from conjur.constants import PASSWORD_COMPLEXITY_CONSTRAINTS_MESSAGE
+from conjur.errors import InvalidPasswordComplexityException, InvalidOperationException, \
+    OperationNotCompletedSuccessfullyException
+
+
+class UserController():
+    """
+    UserController
+
+    This class represents the Presentation Layer for the User command.
+    """
+    def __init__(self, user_logic, user_resource_data):
+        self.user_logic=user_logic
+        self.user_resource_data=user_resource_data
+
+    def rotate_api_key(self):
+        """
+        Method that calls to rotate api key logic to perform the rotation
+        """
+        try:
+            # pylint: disable=line-too-long
+            resource_to_update, new_api_key = self.user_logic.rotate_api_key(self.user_resource_data.user_id)
+            sys.stdout.write(f"Successfully rotated API key for '{resource_to_update}'. " \
+                             f"New API key is: {new_api_key}\n")
+        except InvalidOperationException as invalid_operation:
+            raise InvalidOperationException(f"Error: {invalid_operation}") from invalid_operation
+        except OperationNotCompletedSuccessfullyException as operation_not_completed:
+            # pylint: disable=line-too-long
+            raise OperationNotCompletedSuccessfullyException("Error: Failed to run command to completion leaving " \
+                                                             "the CLI in an unstable state. Log in again or " \
```

Comment:
as a rule of thumb, errors should not contain instructions.
If you want to write to stdout, do it.
I don't see the value in wrapping an exception with the same exception without adding important information.

Reply:
I want to perc this exception up to the `cli.py` because there is the following catch
```
except Exception as error:
    logging.debug(traceback.format_exc())
    sys.stdout.write(f"{str(error)}\n")
    sys.exit(1)
```
which takes care of printing the backtrace in debug mode and formatting.

Reply:
I still think my suggestion stands:
> If you want to write to stdout, do it.

And so is my comment:
> I don't see the value in wrapping an exception with the same exception without adding important information.

So you can throw the same exception (no wrapping), just write to stdout whatever that you want to write.

Reply:
as we spoke about, I am throwing this message in the actual exception body